composables/useProfileImage.ts

```diff
@@ -0,0 +1,65 @@
+import { PROFILE_PLACEHOLDER_SOURCE } from "@/utils/constants/profilePlaceholderSource";
+
+export const useProfileImage = () => {
+  const client = useSupabaseClient();
+  const user = useSupabaseUser();
+
+  const profileImageSource = useProfileImageSource();
+
+  const profileImageError = ref("");
+  const profileImageName = ref("");
+
+  const setProfileImageSource = async () => {
+    if (profileImageName.value) {
+      try {
+        const { getPrivateUrl } = useRetrievePrivateFileUrl(
+          profileImageName.value
+        );
+        const link = await getPrivateUrl();
+        profileImageSource.value = link;
+      } catch (e) {
+        profileImageSource.value = PROFILE_PLACEHOLDER_SOURCE;
+      }
+    } else {
+      profileImageSource.value = PROFILE_PLACEHOLDER_SOURCE;
+    }
+  };
+
+  const checkProfileImage = async () => {
+    const { data, error, status } = await client
+      .from("profiles")
+      .select("avatar")
+      .eq("id", user.value?.id)
+      .single();
+
+    if (error && status !== 406) {
```

Comment:
status 406 is when there is no value in the avatar field - I don't want the error to be threw then so that it is excluded

Reply:
IGNORED_ERROR_CODE constant was added